### PR TITLE
Rename flux package and methods

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

#2755 

*Description of changes:*

Rename Flux package and methods to something simpler and clear. Remove concept "addon" from Flux code.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/gitops/flux/flux.go [hunk 1]
```diff
@@ -53,35 +53,35 @@ const (
...
 		retrier:   retrier.NewWithMaxRetries(maxRetries, backOffPeriod),
 	}
 }
 
-// Flux is an interface that abstracts the basic commands of flux executable.
-type Flux interface {
-	// BootstrapToolkitsComponentsGithub bootstraps toolkit components in a GitHub repository.
-	BootstrapToolkitsComponentsGithub(ctx context.Context, cluster *types.Cluster, fluxConfig *v1alpha1.FluxConfig) error
+// Client is an interface that abstracts the basic commands of flux executable.
+type Client interface {
```
Comment (mitalipaygude) [inline pkg/gitops/flux/flux.go:73]: nit: should we rename it to FluxClient?

Comment (jiayiwang7) [reply pkg/gitops/flux/flux.go:73]: I think it's ok since out of this pkg we will call `flux.Client` instead of `flux.FluxClient` which is duplicate

Comment (danbudris) [reply pkg/gitops/flux/flux.go:73]: package name is Flux, so Client is the go-prefered name in this case

**Pull request merged**